### PR TITLE
fix(binance): parse deposit status, fix #17218

### LINKED ISSUE #17218: binanceus fetch_deposits does not parse statuses correctly
### Operating System

ubuntu

### Programming Languages

python

### CCXT Version

3.0.13

### Description

When fetching deposits from binanceus using the `fetch_deposits` method, I am expecting that the `status` key will be one of `ok`, `failed`, `canceled`, `pending` in accordance with the types documented in https://github.com/ccxt/ccxt/wiki/Manual#deposit-and-withdrawal-networks

However, the status key that is returned is the unparsed status key originally returned by binanceus, which in this case is a string integer.

### Code

```
  

```

### PR DESCRIPTION
**Pull request by pcriadoperez**

### CODE REVIEW

File: ts/src/binance.ts [hunk 1]
```diff
@@ -4967,7 +4967,11 @@ export default class binance extends Exchange {
                 'Refund Failed': 'failed',
             },
         };
-        const statuses = this.safeValue (statusesByType, type, {});
+        const statuses = this.safeValue (statusesByType, type, {
```
Comment (carlosmiei) [inline ts/src/binance.ts:4970]: @pcriadoperez when will we fallback to the default value? isn't easier to add a new category to `statusByType`?

Comment (pcriadoperez) [reply ts/src/binance.ts]: You are right. I changed to add the type in withdrawals and deposit, so then it also gets the correct status type, I think it's a cleaner solution to what I had before.

File: ts/src/binance.ts [hunk 2]
```diff
@@ -4815,6 +4815,9 @@ export default class binance extends Exchange {
             //     }
             //   ]
         }
+        for (let i = 0; i < response.length; i++) {
+            response[i]['type'] = 'deposit';
```
Comment (carlosmiei) [inline ts/src/binance.ts:4819]: by doing this we're modifying the original response which we avoid at all costs,  is this really necessary?  (sorry if I misled you initially I was not suggesting it)

Comment (pcriadoperez) [reply ts/src/binance.ts:4819]: I agree it's not the best, but it's the only way I saw to pass a parameter to `parseTransactions` as we don't have params, and to let it know if it's parsing deposits or transactions. What solution would you recommend if not?

**Comment by carlosmiei** [comment]
LGTM ✅

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**